### PR TITLE
SI-6386 typed existential type tree's original now have tpe set

### PR DESCRIPTION
**Pull request by u-abramchuk**
Tree reification fails for ExistentialTypeTree. The reason is that the
tree passed for reification (see reifyTree at GetTrees.scala) must have
not null tpe (see reifyBoundType at GenTrees.scala), which is not true
in the case of ExistentialTypeTree.

Why is it so? The tree passed to reifyTree was obtained in the reshape
phase of reificationusing using original TypeTrees that represent pre-
typer representation of a type. The problem is that original's tpe for
ExistentialTypeTree is not set.

So the solution to the issue is to create ExistentialTypeTree's original
in a such way that is has actual tpe set.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -3887,8 +3887,10 @@ trait Typers extends Modes with Adaptations with Tags {
         if (vd.symbol.tpe.isVolatile)
           AbstractionFromVolatileTypeError(vd)
       val tpt1 = typedType(tree.tpt, mode)
-      existentialTransform(tree.whereClauses map (_.symbol), tpt1.tpe)((tparams, tp) =>
-        TypeTree(newExistentialType(tparams, tp)) setOriginal tree
+      existentialTransform(tree.whereClauses map (_.symbol), tpt1.tpe)((tparams, tp) => {
+          val original = ExistentialTypeTree(tpt1.asInstanceOf[TypeTree].original, tree.whereClauses)
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: Please, no blind casts.

**Comment by u-abramchuk** [comment]
fixed

**Comment by paulp** [comment]
It's failing because you're creating an unpositioned tree. You want to do something like

```
atPos(tree.pos)(ExistentialTypeTree(tpt.original, tree.whereClauses))
```

**Comment by u-abramchuk** [comment]
Thank you for the hint. Fixed.

**Comment by paulp** [comment]
LGTM

**Pull request merged**